OTAnalytics/plugin_prototypes/track_visualization/track_viz.py

```diff
@@ -519,8 +518,16 @@ def _plot_dataframe(self, track_df: DataFrame, axes: Axes) -> None:
             alpha=self._alpha,
             ax=axes,
             palette=self._color_palette_provider.get(),
-            legend=self._enable_legend,
         )
+        if self._enable_legend:
+            print("creating legend")
+            legend = axes.legend(loc="upper right")
+            legend.set_alpha(1)
+        else:
+            # Somehow enabling a legend from previous call to this method persists.
```

Comment:
Is this (maybe) because the axes persist, too?

Reply:
I just had a look at the code. The `track_viz.MatplotlibTrackPlotter.plot`  method creates the axes via `_create_figure` and `_create_axes` methods. Nothing indicates that existing axes are being used, at least to me 🤔

Reply:
Hm ok, anyway, we have this in mind in case of plotting issues :)